inventory.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+type InventoryAttribute struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+type InventoryData []InventoryAttribute
```

Comment:
Threat my proposal as example here. I see why we might need structure, so then the example will be more like:
```
type InventoryAttribute struct {
	Name  string      `json:"name"`
	Value interface{} `json:"value"`
}

type InventoryData struct {
	data map[string]InventoryAttribute
}

func (id *InventoryData) Write(p []byte) (n int, err error) {
	r := bufio.NewReader(bytes.NewBuffer(p))
	for {
		data, err := r.ReadBytes('\n')
		if err == io.EOF {
			return len(p), nil
		} else if err != nil {
			return 0, err
		}
		ia, _ := readAttr(data[:len(data)-1])
		if data, ok := id.data[ia.Name]; ok {
			switch data.Value.(type) {
			case string:
				newVal := []string{data.Value.(string), ia.Value.(string)}
				id.data[ia.Name] = InventoryAttribute{ia.Name, newVal}
			case []string:
				newVal := append(data.Value.([]string), ia.Value.(string))
				id.data[ia.Name] = InventoryAttribute{ia.Name, newVal}
			}
			continue
		}
		id.data[ia.Name] = ia
	}
}

func readAttr(p []byte) (InventoryAttribute, error) {
	val := strings.SplitN(string(p), "=", 2)
	//TODO: error handling
	return InventoryAttribute{val[0], val[1]}, nil
}
```

It should generate exact same JSON as your proposal. 
Next, I think there is slight misunderstanding of how to use `MultiReader` and `io.Copy`. As an argument to `MultiReader` we can take `StdoutPipe` from `exec` which should be fine to handle or we can use `bytes.NewBuffer()` which implements `io.Reader` interface.
We can of course also take `Output` as you did and call `func (id *InventoryData) Write(p []byte) (n int, err error)` directly for each script.
As for the last, I guess it is the same issue with existing design too. Since we are not distinguishing which script is producing which output IMHO we should threat those equally (no difference if there are multiple same values from one script or from multiple).

To sum up, I come up with some different proposal as I have seen existing one not very intuitive. I agree, though that you can say the same thing to the one I've proposed...